app/src/main/java/it/niedermann/nextcloud/deck/ui/widget/stack/StackWidgetConfigurationActivity.java

```diff
@@ -0,0 +1,62 @@
+package it.niedermann.nextcloud.deck.ui.widget.stack;
+
+import android.appwidget.AppWidgetManager;
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.appcompat.app.ActionBar;
+
+import it.niedermann.nextcloud.deck.DeckLog;
+import it.niedermann.nextcloud.deck.model.Account;
+import it.niedermann.nextcloud.deck.ui.PickStackActivity;
+
+public class StackWidgetConfigurationActivity extends PickStackActivity {
+    private int appWidgetId;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        final ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setTitle("Add stack widget");
+        }
+
+        setResult(RESULT_CANCELED);
+        final Bundle extras = getIntent().getExtras();
+
+        if (extras != null) {
+            appWidgetId = extras.getInt(AppWidgetManager.EXTRA_APPWIDGET_ID,
+                    AppWidgetManager.INVALID_APPWIDGET_ID);
+        }
+
+        if (appWidgetId == AppWidgetManager.INVALID_APPWIDGET_ID) {
+            DeckLog.error("INVALID_APPWIDGET_ID");
+            finish();
+        }
+    }
+
+    @Override
+    protected void onSubmit(Account account, long boardId, long stackId) {
+        final Bundle extras = new Bundle();
+
+        syncManager.addStackWidget(appWidgetId, account.getId(), stackId, false);
+        Intent updateIntent = new Intent(AppWidgetManager.ACTION_APPWIDGET_UPDATE, null,
+                                    getApplicationContext(), StackWidget.class);
+        extras.putSerializable(StackWidget.ACCOUNT_KEY, account);
+        extras.putInt(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId);
+
+        // The `extras` bundle is added to the intent this way because using putExtras(extras) 
+        // would have the OS attempt to reassemle the data and cause a crash 
+        // when it finds classes that are only known to this application.
+        updateIntent.putExtra(StackWidget.BUNDLE_KEY, extras);
```

Comment:
Android is really a bitch, isn't it? I even tried it with `Parcelable` instead of `Serializable`, but not even that worked....

However, we only have to use this strange way in the `StackWidgetConfigurationActivity`. For the intent which starts the `EditActivity` we can use the normal way (I already did the changes and pushed them to this branch).